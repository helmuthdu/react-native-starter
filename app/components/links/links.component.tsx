import * as Samples from '@expo/samples';
import { Body, Button, Container, Content, Header, Icon, Left, Right, Title } from 'native-base';
import React from 'react';
import { ScrollView } from 'react-native';
import { NavigationRoute, NavigationScreenProp } from 'react-navigation';
import { styles } from './links.style';

export const Links = (props: { navigation: NavigationScreenProp<NavigationRoute> }) => (
  <Container style={styles.container}>
    {props.navigation.openDrawer && (
      <Header>
        <Left>
          <Button transparent>
            <Icon active name="menu" onPress={() => props.navigation.openDrawer()} />
          </Button>
        </Left>
        <Body>
          <Title>Links</Title>
        </Body>
        <Right />
      </Header>
    )}
    <Content>
      <ScrollView style={styles.container}>
        {/* Go ahead and delete ExpoLinksView and replace it with your
         * content, we just wanted to provide you with some helpful links */}
        <Samples.ExpoLinksView />
      </ScrollView>
    </Content>
  </Container>
);
