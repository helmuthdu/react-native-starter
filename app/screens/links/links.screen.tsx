import React, { Component } from 'react';
import { NavigationRoute, NavigationScreenProp } from 'react-navigation';
import { Links } from '../../components/links/links.component';

type Props = {
  navigation: NavigationScreenProp<NavigationRoute>;
};

export default class LinksScreen extends Component<Props> {
  public render() {
    return <Links navigation={this.props.navigation} />;
  }
}
