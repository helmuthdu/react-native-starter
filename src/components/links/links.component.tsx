import React from 'react';
import * as Samples from '@expo/samples';
import { NavigationRoute, NavigationScreenProp } from 'react-navigation';
import { Body, Button, Container, Content, Header, Icon, Left, Right, Title } from 'native-base';
import { styles } from './links.style';
import { ScrollView } from 'react-native';

export const Links = (props: { navigation: NavigationScreenProp<NavigationRoute> }) => (
  <Container style={styles.container}>
    {props.navigation.openDrawer && (
      <Header>
        <Left>
          <Button transparent>
            <Icon active name="menu" onPress={() => props.navigation.openDrawer()} />
          </Button>
        </Left>
        <Body>
          <Title>Links</Title>
        </Body>
        <Right />
      </Header>
    )}
    <Content>
      <ScrollView style={styles.container}>
        {/* Go ahead and delete ExpoLinksView and replace it with your
         * content, we just wanted to provide you with some helpful links */}
        <Samples.ExpoLinksView />
      </ScrollView>
    </Content>
  </Container>
);
