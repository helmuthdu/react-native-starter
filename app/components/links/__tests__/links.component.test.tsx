import React from 'react';
import 'react-native';
// Note: test renderer must be required after react-native.
import * as renderer from 'react-test-renderer';
import { Links } from '../links.component';

const navigation = { navigate: jest.fn() } as any;

it('renders correctly', () => {
  const tree = renderer.create(<Links navigation={navigation} />).toJSON();
  expect(tree).toMatchSnapshot();
});
