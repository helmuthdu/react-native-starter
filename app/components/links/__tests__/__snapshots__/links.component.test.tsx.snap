// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
        "height": 1334,
      },
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
      },
    ]
  }
>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentContainerStyle={
      Array [
        Object {
          "padding": undefined,
        },
        undefined,
      ]
    }
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutomaticScroll={true}
    enableOnAndroid={false}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    getScrollResponder={[Function]}
    handleOnScroll={[Function]}
    keyboardDismissMode="interactive"
    keyboardOpeningTime={250}
    keyboardShouldPersistTaps="handled"
    keyboardSpace={0}
    onScroll={[Function]}
    resetKeyboardSpace={[Function]}
    resetScrollToCoords={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    scrollEventThrottle={1}
    scrollForExtraHeightOnAndroid={[Function]}
    scrollIntoView={[Function]}
    scrollToEnd={[Function]}
    scrollToFocusedInput={[Function]}
    scrollToPosition={[Function]}
    showsVerticalScrollIndicator={true}
    style={
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
      }
    }
    update={[Function]}
    viewIsInsideTabBar={false}
  >
    <View>
      <RCTScrollView
        style={
          Object {
            "backgroundColor": "#fff",
            "flex": 1,
          }
        }
      >
        <View>
          <View>
            <Text
              style={
                Object {
                  "fontSize": 16,
                  "marginBottom": 12,
                  "marginLeft": 15,
                  "marginTop": 9,
                }
              }
            >
              Resources
            </Text>
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "backgroundColor": "#fdfdfd",
                  "borderBottomColor": "#EDEDED",
                  "borderBottomWidth": 0.5,
                  "opacity": 1,
                  "paddingHorizontal": 15,
                  "paddingVertical": 15,
                }
              }
            >
              <View
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "marginRight": 9,
                    }
                  }
                >
                  <Image
                    fadeDuration={0}
                    resizeMode="contain"
                    source={1}
                    style={
                      Object {
                        "height": 20,
                        "marginTop": 1,
                        "width": 20,
                      }
                    }
                  />
                </View>
                <View>
                  <Text
                    style={
                      Object {
                        "fontSize": 15,
                        "marginTop": 1,
                      }
                    }
                  >
                    Read the Expo documentation
                  </Text>
                </View>
              </View>
            </View>
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "backgroundColor": "#fdfdfd",
                  "borderBottomColor": "#EDEDED",
                  "borderBottomWidth": 0.5,
                  "opacity": 1,
                  "paddingHorizontal": 15,
                  "paddingVertical": 15,
                }
              }
            >
              <View
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "marginRight": 9,
                    }
                  }
                >
                  <Text />
                </View>
                <View>
                  <Text
                    style={
                      Object {
                        "fontSize": 15,
                        "marginTop": 1,
                      }
                    }
                  >
                    Ask a question on the Expo forums
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </RCTScrollView>
</View>
`;
